src/controllers/expenses.controller.js

```
@@ -0,0 +1,125 @@
+const { expensesService } = require('../services/expenses.service');
+const { usersService } = require('../services/users.service');
+
+const getExpenses = async (req, res) => {
+  const { userId, categories, from, to } = req.query;
+  const expenses = await expensesService.getExpenses();
+
+  let filteredExpences = expenses;
+
+  if (userId) {
+    filteredExpences = filteredExpences.filter(
+      (expense) => expense.userId === Number(userId),
+    );
```

When filtering by `userId` you coerce with `Number(userId)` and then compare. If `userId` is non-numeric this becomes NaN and the comparison silently fails. You must validate `userId` (e.g., `if (userId && Number.isNaN(Number(userId))) return res.status(400).json({ message: 'userId must be a number' })`) before performing the filter.